Updates to support calculating the I18N score

ehoogerbeets opened the pull request:
- add FileStats class to keep track of number of files, lines, and modules
- returned from the parser as a field in the IntermediateRepresentation

ehoogerbeets committed:
Add a stats parameter to the Parser object

- should contain stats about the source file that was parsed
    - number of lines, modules, etc.

ehoogerbeets committed:
Add FileStats class

- keep track of number of files, lines, and modules
- returned by the parser in the intermediate representation

ehoogerbeets committed:
Update README and package version

wadimw reviewed (commented):
https://github.com/iLib-js/i18nlint/pull/36#issuecomment-1538065943

ehoogerbeets committed:
Add charNumber field to a Result as well

ehoogerbeets committed:
Make sure props are defined

... before copying them into the current instance

ehoogerbeets committed:
Fixed to use the right parameter name

ehoogerbeets committed:
Add endLineNumber and endCharNumber to Result

- optionally indicates the end of where the problem occurred in the
  source file

ehoogerbeets committed:
Added rel note

ehoogerbeets committed:
Add support for bytes in file stats

ehoogerbeets committed:
Export FileStats class from the index

ehoogerbeets committed:
Add unit test for endLineNumber and endCharNumber

ehoogerbeets committed:
Fix a bug with empty stats

- adding other stats to empty stats objects results in NaN for
  most counts

ehoogerbeets committed:
Update dependencies

ehoogerbeets committed:
Update documentation
